src/sinks/datadog/metrics/encoder.rs

```diff
@@ -183,10 +187,20 @@ impl DatadogMetricsEncoder {
             compressed_limit,
             state: EncoderState::default(),
             log_schema: log_schema(),
+            origin_product_value: determine_origin_product_value(),
         })
     }
 }
 
+fn determine_origin_product_value() -> u32 {
```

Comment:
Nit: It seems like this could be `const` but maybe the compiler isn't smart enough.

Reply:
Tried but `unwrap_or` and `map` are not const so the compiler does complain about that.

Reply:
This is a good candidate for a `once_cell::sync::Lazy`.